node_modules/oae-authentication/lib/api.js

```diff
@@ -307,12 +308,19 @@ var _getOrCreateUser = function(ctx, loginId, displayName, opts, callback) {
                     opts.emailVerified = true;
                 }
 
-                createUser(ctx, loginId, displayName, opts, function(err, user) {
+                var shouldCheckEmail = (opts.email && !opts.authoritative);
+                OaeUtil.invokeIfNecessary(shouldCheckEmail, validateEmailBelongsToTenant, ctx, opts.email, function(err) {
```

Comment:
Any reason not to do the same thing we would do if the non-authoritative external login method provided an email that didn't match the tenant's email domain ?